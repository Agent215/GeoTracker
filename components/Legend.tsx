import CloudsLegend from '../assets/openWeatherKeys/cloudsKey.svg'
import PrecipitationKey from '../assets/openWeatherKeys/precipitationKey.svg'
import PressureKey from '../assets/openWeatherKeys/pressureKey.svg'
import TemperatureKey from '../assets/openWeatherKeys/temperature.svg'
import WindKey from '../assets/openWeatherKeys/windKey.svg';

import React from 'react';




/* This function creates the correct weather Legeend using an SVG file,
then returns this component. It takes two props, category and size to display. */
const WeatherLegend = (props) => {

    /* add categories to ENUM object, not sure how to extract the width and height here,
    there is probably a better way to do this so we dont have to repeat the height and width everytime.*/
    const categorySVG = {
        ["clouds"]: <CloudsLegend style={{ marginTop: 50 }} width={props.size} />,
        ["precipitation"]: <PrecipitationKey style={{ marginTop: 50 }} width={props.size} />,
        ["pressure"]: <PressureKey style={{ marginTop: 50 }} width={props.size} />,
        ["temp"]: <TemperatureKey style={{ marginTop: 50 }} width={props.size} />,
        ["wind"]: <WindKey style={{ marginTop: 50 }} width={props.size} />,
        ["none"]: null,

    }

    const weatherKey = categorySVG[props.category];

    if (!weatherKey) {
        console.log("Legend.tsx/WeatherLegend() - No weather of type" + props.category + " found");
        return null
    };

    return (weatherKey);

}

export default WeatherLegend;