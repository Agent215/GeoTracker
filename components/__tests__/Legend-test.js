import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Legend from '../../components/Legend';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockedNavigator from '../MockedNavigator'


const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

enzyme.configure({ adapter: new Adapter() });
// mock of props to pass disaster card
const size= 22;
const category = "clouds"


describe('Testing Legend Component', () => {
  const initialState = { output: "null" }
  const mockStore = configureStore()
  let store, wrapper
  /*
  Check Legend against snapshot. run jest -u to update snapshot
  */
  it(`Legend renders correctly`, () => {
    store = mockStore(initialState)

    const comp = <Provider store={store}><Legend size={size} category={category} /></Provider>
    const tree = renderer.create(comp).toJSON();

    expect(tree).toMatchSnapshot();
  });
  /**
   * Legend should not be null
   */
  it(`Legend should not be null`, () => {
    const comp = <Provider store={store}><Legend size={size} category={category}/></Provider>
    const tree = renderer.create(comp).toJSON();

    expect(tree).toBeTruthy();
  });

 
})
