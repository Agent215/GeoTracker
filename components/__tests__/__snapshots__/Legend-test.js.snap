// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Legend Component Legend renders correctly 1`] = `
<SvgMock
  style={
    Object {
      "marginTop": 40,
    }
  }
  width={22}
/>
`;
